README.md

```diff
@@ -1,7 +1,133 @@
+
+
+## Using lungair-fhir-server
+
+With the initial setup finished, there are two ways to use
+`lungair-fhir-server`:
+1. Use one of the existing data sources, such as the random data generator used for testing, or the data source that takes NICU data from a downloaded MIMIC-III dataset.
+2. Create your own `PatientDataSource` to populate your own custom patient and observation data into a FHIR server.
+
+### Random data
+
+This approach can be used to test basic functionality by populating a FHIR server with random data:
+
+```sh
+python populate_fhir_server.py --json_file ./data_sources/random.json
+    --fhir_server http://localhost:3000/hapi-fhir-jpaserver/fhir/    
+```
+
+The amount of data to generate can be configured in [data_sources/random.json](data_sources/random.json).
+
+### MIMIC-III
+
+This approach uses a downloaded [MIMIC-III](https://physionet.org/content/mimiciii/1.4/) dataset to populate a FHIR server with NICU patients and ventilator observations. Note that the MIMIC-III dataset requires credentialed access on PhysioNet.
+
+1. Get access to and download [MIMIC-III](https://physionet.org/content/mimiciii/1.4/).
+2. Configure [data_sources/MIMIC-III.json](data_sources/MIMIC-III.json) to point to the location of your downloaded data and the MIMIC-III schema files:
+    ```json
+    "args":
+    {
+        "mimic3_data_dir": "path/to/mimic3/data/dir",
+        "mimic3_schemas_dir": "path/to/mimic3/schema/dir"
+    }
+    ```
+    The schema files are provided in this repository [here](mimic3-schemas).
+3. Run `populate_fhir_server.py`:
+    ```sh
+    python populate_fhir_server.py --json_file ./data_sources/mimic3.json
+    	--fhir_server http://localhost:3000/hapi-fhir-jpaserver/fhir/    
+    ```
+
+
+### Custom data source
+
+To populate the FHIR server with your own data, you will need to write a bit of python.
+The procedure is to subclass the `PatientDataSource`, `Patient`, and `Observation` classes in [data_sources/patient_data_source.py](data_sources/patient_data_source.py), specifying how the patient and observation data should be created.
+
+The following steps create a minimal working data source:
+1. Create a new python file in for your new data source, importing `PatientDataSource`, `Patient`, and `Observation` from [data_sources/patient_data_source.py](data_sources/patient_data_source.py).
+2. Inside the new python file, create two implementations of `Observation`
+and `PatientDataSource` from `patient_data_source.py`.
+3. Implement `__init__` for your `PatientDataSouce` implementation
+This method should handle importing your data source into python.
+4. Implement `create_patient` and `get_patient_observations` for
+your `PatientDataSource` implementation. Read the doc strings in
+`patient_data_source.py` for more information.
+5. Implement `get_observation_type` and `get_value` for your
```

Comment:
Looking at this, I'm starting to think an example would be a lot more useful than a step-by-step description like this.

I made up some example data: [example.csv](https://github.com/KitwareMedical/lungair-fhir-server/files/9823718/example.csv)
We can show the table in markdown form, like "suppose you have this data and you want to populate a fhir server with it"

Let's add the observation type for body weight:
```json
"bodyweight":
{
    "display_string": "body weight",
    "unit_code": "kg",
    "loinc_code": "29463-7"
}
```
(see [here](https://loinc.org/29463-7/) for the loinc code description)


and then instead of step-by-step instructions we can just show how you would subclass the appropriate classes, load the data with pandas, implement the correct methods, make a json, etc.

What do you think?

Reply:
I am a bit confused here. Are we adding body weight observation type as a type we implement or is this just for the example?

Reply:
my intention was to add it to or observation types json so that we support it,  might as well

Reply:
In either case, I think it would be best to add another file showing a walk-through of all this information. At the end of the basic steps, we can say a more in depth explanation/walk-through can be found here. I think having an in-depth explanation of how to add a data source inside the README will make it cluttered.

We could even change the adding custom data sources to say something along the lines of "It is possible to add a custom data source by creating implementations of `Observation` and `PatientDataSource` from `patient_data_source.py`].  A more in-depth explanation can be found here." This will allow us to go into as much detail as we want inside the walk-through without extending the length of the README to a ridiculous size.

Reply:
Hmm sure we can separate it out. I like that!

How about we make a directory in this repo called "example" and put a README.md in there, example.csv that I just shared, and also put in `example_data_source.py` which defines `ExampleDataSource` and `ExampleObservation` and `ExamplePatient`, and finally `example_data_source.json`.

Then we can link to that folder here